From 4c5401920d1cd094e87c5cb99f7ac4e3e9cbbf44 Mon Sep 17 00:00:00 2001
From: Lasse Brun <bruners@gmail.com>
Date: Sun, 18 May 2014 05:28:09 +0200
Subject: [PATCH] Compile fix for Linux 3.14

Source: Upstream forum
Upstream: Posted on forum
Reason: Fails to build on 3.14 kernels

---
 setup/Linux/oss/build/osscore.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/setup/Linux/oss/build/osscore.c b/setup/Linux/oss/build/osscore.c
index d79c6a6..f31630f 100644
--- a/setup/Linux/oss/build/osscore.c
+++ b/setup/Linux/oss/build/osscore.c
@@ -472,7 +472,9 @@ oss_get_pid (void)
 int
 oss_get_uid (void)
 {
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,29)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,14,0)
+  return __kuid_val(current->cred->uid);
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,29)
   return current->cred->uid;
 #else
   return current->uid;
-- 
1.9.3

